bigquery/tests/test_query_builder.py

```diff
@@ -3,6 +3,40 @@
 
 unittest.TestCase.maxDiff = None
 
+class TestGenerateFunctions(unittest.TestCase):
+    def test_single_function(self):
+        from bigquery.query_builder import generate_formatter
+
+        self.assertEqual(generate_formatter('MAX'), 'MAX')
+
+    def test_single_function_with_args(self):
+        from bigquery.query_builder import generate_formatter
+
+        argument_list = list()
+        argument_list.append(10)
+        argument_list.append('test')
+        self.assertEqual(generate_formatter('LEFT',
+            arguments=argument_list), 'LEFT:10,test')
+
+    def test_nested_functions(self):
+        from bigquery.query_builder import generate_formatter
+
+        integer_function = generate_formatter('INTEGER')
+        max_integer_function = generate_formatter('MAX',
+                inner_function=integer_function)
+        self.assertEqual('INTEGER-MAX', max_integer_function)
+
+    def test_nesting_with_args(self):
+        from bigquery.query_builder import generate_formatter
+        
+        argument_list = list()
```

Comment:
Cool thing about python: list construction can be declarative -- no need to make the object and mutate it. Something like:
`foo = [bang, bar]`